### PR TITLE
Rhb 31

### PR DESCRIPTION
**Pull request by Avaxapsys**

### CODE REVIEW

File: DevEdu.API/DevEdu.API/Configuration/MapperProfile.cs [hunk 1]
```diff
@@ -66,7 +66,9 @@ private void CreateMappingFromDto()
             CreateMap<CommentDto, CommentInfoOutputModel>()
                 .ForMember(dest => dest.Date, opt => opt.MapFrom(src => src.Date.ToString(_dateFormat)));
             CreateMap<CourseTopicDto, CourseTopicOutputModel>();
-            CreateMap<CourseDto, CourseInfoBaseOutputModel>();
+            //CreateMap<CourseDto, CourseInfoBaseOutputModel>();
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Configuration/MapperProfile.cs:69]: please remove this line if it is not needed

File: DevEdu.API/DevEdu.Business.Tests/AuthenticationServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DevEdu.Business.Tests
+{
+    class AuthenticationServiceTests
+    {
+        private AuthenticationService _auth;
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business.Tests/AuthenticationServiceTests.cs:14]: please rename it to _sut to keep it consistent

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs [hunk 3]
```diff
@@ -20,11 +20,136 @@ public class CourseServiceTests
...
+        [Test]
+        public void AddCourse_CourseInput_CourseCreated() 
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+
+            _courseRepositoryMock.Setup(x => x.AddCourse(courseDto)).Returns(courseId);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:38]: can we have it in Setup as well?

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs [hunk 4]
```diff
@@ -20,11 +20,136 @@ public class CourseServiceTests
...
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var dto = sut.GetCourse(courseId);
+
+            //Than
+            Assert.AreEqual(courseDto, dto);
+            _courseRepositoryMock.Verify(x => x.GetCourse(1));
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:87]: please add Times in each Verify method
and please check that tose methods are not being called:
image

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs [hunk 5]
```diff
@@ -20,11 +20,136 @@ public class CourseServiceTests
...
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var dto = sut.GetFullCourseInfo(courseId);
+
+            //Than
+            Assert.AreEqual(courseDto, dto);
+            _courseRepositoryMock.Verify(x => x.GetCourse(courseId));
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:106]: please add verification for those methods:
image

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs [hunk 6]
```diff
@@ -20,11 +20,136 @@ public class CourseServiceTests
...
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+
+            //When
+            var actualCourseDto = sut.GetCourses();
+
+            //Then
+            Assert.AreEqual(courseDto, actualCourseDto);
+            _courseRepositoryMock.Verify(x => x.GetCourses(), Times.Once);
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:127]: please add verification for methods:
image

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs [hunk 7]
```diff
@@ -20,11 +20,136 @@ public class CourseServiceTests
...
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+
+            //When
+            var actualCourseDto = sut.UpdateCourse(courseId, courseDto);
+
+            //Then
+            Assert.AreEqual(updCourseDto, actualCourseDto);
+            _courseRepositoryMock.Verify(x => x.UpdateCourse(courseId, courseDto), Times.Once);
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:150]: please add verification for methods:
image

File: DevEdu.API/DevEdu.Business.Tests/Data/CourseData.cs [hunk 8]
```diff
@@ -12,6 +8,152 @@ public class CourseData
         public CourseData()
         {
 
+        }
+        public const int ExpectedAffectedRows = 1;
+        public const int CourseId = 1;
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business.Tests/Data/CourseData.cs:13]: it's a bad practice to have such values as consts
it's better to declare them in each test method separately

File: DevEdu.API/DevEdu.Business.Tests/Data/GroupData.cs [hunk 9]
```diff
@@ -5,5 +5,6 @@ public class GroupData
         public const int ExpectedAffectedRows = 1;
         public const int GroupId = 1;
         public const int MaterialId = 1;
+        public const int LessonId = 1;
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business.Tests/Data/GroupData.cs:8]: it's a bad practice to have such values as consts
it's better to declare them in each test method separately

File: DevEdu.API/DevEdu.API/Configuration/MapperProfile.cs (hunk 1 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.API/Configuration/MapperProfile.cs:69]: +

File: DevEdu.API/DevEdu.Business.Tests/AuthenticationServiceTests.cs (hunk 2 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business.Tests/AuthenticationServiceTests.cs:14]: +

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs (hunk 3 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:38]: +

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs (hunk 4 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:87]: +

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs (hunk 5 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:106]: +

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs (hunk 6 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:127]: +

File: DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs (hunk 7 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs:150]: +

File: DevEdu.API/DevEdu.Business.Tests/Data/CourseData.cs (hunk 8 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business.Tests/Data/CourseData.cs:13]: +

File: DevEdu.API/DevEdu.Business.Tests/Data/GroupData.cs (hunk 9 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business.Tests/Data/GroupData.cs:8]: 1

**Pull request merged**